src/main/java/duke/tasks/Task.java

```diff
@@ -0,0 +1,171 @@
+package duke.tasks;
+
+import duke.DukeException;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+/**
+ * Base Task class which provides shared functionality such as description, task type and status
+ */
+public class Task {
+    private final String description;
+    private final TaskType type;
+    private TaskStatus status;
+
+    /**
+     * Constructor for Task class
+     *
+     * @param description of task
+     * @param type        of task
+     * @see TaskStatus
+     */
+    public Task(String description, TaskType type) {
+        this.description = description;
+        this.type = type;
+        this.status = TaskStatus.PENDING;
+    }
+
+    /**
+     * Constructor for Task class with completion status specified
+     *
+     * @param description of task
+     * @param type        of task
+     * @param completion  <code>true</code> if the event is completed, <code>false</code> otherwise
+     * @see TaskStatus
+     */
+    public Task(String description, TaskType type, Boolean completion) {
+        this.description = description;
+        this.type = type;
+        this.status = completion ? TaskStatus.COMPLETED : TaskStatus.PENDING;
+    }
+
+    /**
+     * Parse commands and creates Todo, Event or Deadline objects based on the type of command
+     *
+     * @param command to be parsed
+     * @return Respective Task object based on command issued
+     * @throws DukeException if incomplete or unrecognizable command issued
+     */
+    public static Task parseTask(String command) throws DukeException {
+        try {
+            String[] inputSplit = command.split(" ", 2);
+            String entryType = inputSplit[0];
+            String description = inputSplit[1].strip();
+            switch (entryType) {
+            case "todo":
+                return Todo.parseTodo(description);
+            case "event":
+                return Event.parseEvent(description);
+            case "deadline":
+                return Deadline.parseDeadline(description);
+            default:
+                throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("☹ OOPS!!! I'm sorry, but I the description of the task cannot be empty :-(");
+        }
+    }
+
+    /**
+     * Parse record read from disk by Storage objects and returns the corresponding Task object
+     *
+     * @param fullRecord Single entry read from file
+     * @return Corresponding Task object
+     */
+    public static Task parseRecord(String fullRecord) {
+        String[] recordSplit = fullRecord.split("\\|");
```

Comment:
Perhaps a plural could be used for the variable name of this string array?

Reply:
Thank you for your comment, good suggestion!